src/Patterns.jl

```diff
@@ -115,6 +115,8 @@ PatExpr(iscall, op, args::Vector) = PatExpr(iscall, op, maybe_quote_operation(op
 
 isground(p::PatExpr)::Bool = p.isground
 
+Base.:(==)(x::PatExpr, y::PatExpr) = x.head==y.head && x.children==y.children
```

Comment:
needed to make `rule == rule` work. I am not sure if we should check for something else in the `PatVar`s?

Reply:
You can adjust by using the hashes.